src/components/NewProjectFlow/flow/ProjectEntitiesFiltersPopover.tsx

```diff
@@ -0,0 +1,251 @@
+
+  useEffect(() => {
+    if (flowState === 'accessions') {
+      setEntitySpecificOptions(ACCESSION_2_STATES);
+    } else if (flowState === 'batches') {
+      setEntitySpecificOptions(nurseries.map((nursery) => nursery.id));
+    }
+  }, [selectedOrganization, flowState, nurseries]);
+
+  const EntitySpecificFilterMultiSelect = useMemo(() => {
+    let label = '';
+    let initialSelection: (number | string)[] = [];
+    let filterKey = '';
+    let renderOption = (value: number | string) => `${value}`;
+
+    if (flowState === 'accessions') {
+      label = strings.STATUS;
+      initialSelection = filters.statuses ?? [];
+      filterKey = 'statuses';
+    } else if (flowState === 'batches') {
+      label = strings.NURSERY;
+      initialSelection = filters.nurseryIds ?? [];
+      filterKey = 'nurseryIds';
+      renderOption = (nurseryId: number | string) => nurseries.find((nursery) => nursery.id === nurseryId)?.name || '';
+    }
+
+    const isOpen = Boolean(anchorEls[filterKey]);
+
+    return (
+      <div>
+        <div className={classes.dropdown} onClick={(event) => handleClick(event, filterKey)}>
+          <Typography>{label}</Typography>
+          <Icon name={isOpen ? 'chevronUp' : 'chevronDown'} className={classes.dropdownIconRight} />
+        </div>
+        {isMobile && isOpen ? (
+          <div className={classes.mobileContainer}>
+            <FilterMultiSelect
+              label={label}
+              initialSelection={initialSelection}
+              onCancel={() => handleClose(filterKey)}
+              onConfirm={(selected) => handleConfirm(filterKey, selected)}
+              options={entitySpecificOptions}
+              renderOption={renderOption}
+            />
+          </div>
+        ) : (
+          <Popover
+            id='pre-exposed-filter-popover'
+            open={isOpen}
+            onClose={handleClose}
+            anchorEl={anchorEls[filterKey]}
+            anchorOrigin={{
+              vertical: 'bottom',
+              horizontal: 'left',
+            }}
+            transformOrigin={{
+              vertical: 'top',
+              horizontal: 'left',
+            }}
+            className={classes.popoverContainer}
+          >
+            <FilterMultiSelect
+              label={label}
+              initialSelection={initialSelection}
+              onCancel={() => handleClose(filterKey)}
+              onConfirm={(selected) => handleConfirm(filterKey, selected)}
+              options={entitySpecificOptions}
+              renderOption={renderOption}
+            />
+          </Popover>
+        )}
+      </div>
+    );
+  }, [
+    anchorEls,
+    classes,
+    entitySpecificOptions,
+    filters.nurseryIds,
+    filters.statuses,
+    flowState,
+    handleClick,
+    handleClose,
+    handleConfirm,
+    isMobile,
+    nurseries,
+  ]);
+
+  const ProjectFilterMultiSelect = useMemo(() => {
```

Comment:
Just curious, is there an extractable component from here that can be reused for both use-cases?

Reply:
I did consider it, I feel like that optimization should be made if we add another filter. I was also thinking about making this even more generic, so the "entity specific filtering configuration" was passed in by the `SelectAccessions|Batches|PlantingSites` view. These components are hyper specific to this tree of functionality and could be made a bit more generic

Reply:
A filter config sounds great 👍

Reply:
Alright take a look now, the entity-specific filters are passed down from the `Select$entity` components. The "all entities" project filter is created within the Search now and we are using a generic filter component. Left open in a way where we should be able to easily add more filters to specific entities if needed.